eclair-core/src/test/scala/fr/acinq/eclair/channel/RestoreSpec.scala

```diff
@@ -0,0 +1,229 @@
+          TestActor.KeepRunning
+        case _ => TestActor.KeepRunning
+      }
+    })
+    aliceListener
+  }
+
+  test("restore channel without configuration change") { f =>
+    import f._
+    val sender = TestProbe()
+    val channelUpdateListener = {
+      val listener = TestProbe()
+      system.eventStream.subscribe(listener.ref, classOf[ChannelUpdateParametersChanged])
+      aliceChannelUpdateListener(listener)
+    }
+
+    // we start by storing the current state
+    assert(alice.stateData.isInstanceOf[DATA_NORMAL])
+    val oldStateData = alice.stateData.asInstanceOf[DATA_NORMAL]
+
+    // we simulate a disconnection
+    sender.send(alice, INPUT_DISCONNECTED)
+    sender.send(bob, INPUT_DISCONNECTED)
+    awaitCond(alice.stateName == OFFLINE)
+    awaitCond(bob.stateName == OFFLINE)
+
+    // and we terminate Alice
+    alice.stop()
+
+    // we restart Alice
+    val newAlice: TestFSMRef[ChannelState, ChannelData, Channel] = TestFSMRef(new Channel(Alice.nodeParams, wallet, Bob.nodeParams.nodeId, alice2blockchain.ref, relayerA.ref, FakeTxPublisherFactory(alice2blockchain)), alicePeer.ref)
+    newAlice ! INPUT_RESTORED(oldStateData)
+
+    newAlice ! INPUT_RECONNECTED(alice2bob.ref, aliceInit, bobInit)
+    bob ! INPUT_RECONNECTED(bob2alice.ref, bobInit, aliceInit)
+    alice2bob.expectMsgType[ChannelReestablish]
+    bob2alice.expectMsgType[ChannelReestablish]
+    alice2bob.forward(bob)
+    bob2alice.forward(newAlice)
+    awaitCond(newAlice.stateName == NORMAL)
+
+    channelUpdateListener.expectNoMessage()
+  }
+
+  test("restore channel with configuration change") { f =>
+    import f._
+    val sender = TestProbe()
+    val channelUpdateListener = {
+      val listener = TestProbe()
+      system.eventStream.subscribe(listener.ref, classOf[ChannelUpdateParametersChanged])
+      aliceChannelUpdateListener(listener)
+    }
+
+    // we start by storing the current state
+    assert(alice.stateData.isInstanceOf[DATA_NORMAL])
+    val oldStateData = alice.stateData.asInstanceOf[DATA_NORMAL]
+
+    // we simulate a disconnection
+    sender.send(alice, INPUT_DISCONNECTED)
+    sender.send(bob, INPUT_DISCONNECTED)
+    awaitCond(alice.stateName == OFFLINE)
+    awaitCond(bob.stateName == OFFLINE)
+
+    // and we terminate Alice
+    alice.stop()
+
+    // we restart Alice with a different configuration
+    val newFees = RelayFees(765 msat, 2345)
+    val newConfig = Alice.nodeParams.copy(relayParams = RelayParams(newFees, newFees, newFees))
+    val newAlice: TestFSMRef[ChannelState, ChannelData, Channel] = TestFSMRef(new Channel(newConfig, wallet, Bob.nodeParams.nodeId, alice2blockchain.ref, relayerA.ref, FakeTxPublisherFactory(alice2blockchain)), alicePeer.ref)
+    newAlice ! INPUT_RESTORED(oldStateData)
+
+    val u1 = channelUpdateListener.expectMsgType[ChannelUpdateParametersChanged]
+    assert(!Announcements.areSameIgnoreFlags(u1.channelUpdate, oldStateData.channelUpdate))
```

Comment:
Maybe assert that we use the correct values as well (765 msat and 2345 ppm)?

Reply:
Done with 290ec5731.